Detectors/TRD/simulation/src/Detector.cxx

```diff
@@ -102,24 +107,33 @@ bool Detector::ProcessHits(FairVolume* v)
     return false;
   }
 
-  // Determine the dectector number
+  // Determine the detector number
   int sector, det;
   // The plane number and chamber number
   char cIdChamber[3];
   cIdChamber[0] = cIdCurrent[2];
   cIdChamber[1] = cIdCurrent[3];
   cIdChamber[2] = 0;
-  // The det-sec number (0 – 29)
+  // The det-sec number (0 - 29)
   const int idChamber = mGeom->getDetectorSec(atoi(cIdChamber));
+  if (idChamber < 0 || idChamber > 29) {
+    LOG(FATAL) << "Chamber ID out of bounds";
+  }
   // The sector number (0 - 17), according to the standard coordinate system
-  TString cIdPath = fMC->CurrentVolPath();
+  TString cIdPath = fMC->CurrentVolOffName(7); // will return BTRD0....BTRD17
   char cIdSector[3];
-  cIdSector[0] = cIdPath[21];
-  cIdSector[1] = cIdPath[22];
+  cIdSector[0] = cIdPath[4];
```

Comment:
I agree with the previous comments that a constant offset into a string could be problematic, and I find it not very readable. Finding the required string or using sscanf should safer and more readable.

Reply:
I think the current constant offset against the volume offset name is safe enough now. Following Sandro's suggestion I would avoid any additional string handling.

Reply:
Note that, compared to before, using `fMC->CurrentVolOffName(7)` will always return a string `BTRD0` and `BTRD17`, so that the constant offset is way safer than before; unless additional geometry levels are attached upstream, which I don't think will be the case. If it is, I added a few comment lines in TRDGeometry.cxx to keep this in mind.